core/main/src/firebolt/handlers/advertising_rpc.rs

```diff
@@ -359,6 +359,14 @@ impl AdvertisingServer for AdvertisingImpl {
         )
         .await;
 
+        let country_code = StorageManager::get_string(&self.state, StorageProperty::CountryCode)
```

Comment:
Looks like we are repeating this code sequence/logic in multiple places.  You may move this to the code that provides `privacy_data` value.